File: backend/src/main/java/com/springbootjwtpostgres/backend/report/Report.java
package com.springbootjwtpostgres.backend.report;

import com.springbootjwtpostgres.backend.basemodels.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import java.util.Date;

@Entity
@EqualsAndHashCode(callSuper = true)
@Data
@NoArgsConstructor
public class Report extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private ReportPeriod reportPeriod;
    private int beforeQuantity;
    private int afterQuantity;
    private double totalPriceIn;
    private double totalPriceOut;
    private Date createdAt;
}

File: frontend/src/app/models/report/Report.ts
import { BaseEntity } from '../base/BaseEntity';
import { EReportPeriod } from './EReportPeriod';

export class Report extends BaseEntity {
  reportPeriod: EReportPeriod;
  beforeQuantity: number;
  afterQuantity: number;
  totalPriceIn: number;
  totalPriceOut: number;

  constructor(
    reportPeriod: EReportPeriod,
    beforeQuantity: number,
    afterQuantity: number,
    totalPriceIn: number,
    totalPriceOut: number
  ) {
    super();
    this.reportPeriod = reportPeriod;
    this.beforeQuantity = beforeQuantity;
    this.afterQuantity = afterQuantity;
    this.totalPriceIn = totalPriceIn;
    this.totalPriceOut = totalPriceOut;
  }
}
